### PR TITLE
Wait for STOP effector to complete before unamanging BrooklynNode

### PR DESCRIPTION
**Pull request by neykov**
Otherwise if there still are running tasks for the entity they will be canceled.

### CODE REVIEW

File: core/src/main/java/brooklyn/entity/basic/BrooklynTaskTags.java [hunk 1]
```diff
@@ -346,6 +347,26 @@ public static boolean isInEffectorTask(Task<?> task, @Nullable Entity entity, @N
         return false;
     }
 
+    @Beta
+    public static Task<?> getEffectorParentTask(Task<?> child, Effector<?> effector) {
```
Comment (ahgittin) [inline core/src/main/java/brooklyn/entity/basic/BrooklynTaskTags.java]: rename as `getClosestEffectorTask()` ?  it's not necessarily the parent.  `closest` is used in the jquery sense.

might be useful eventually to refactor to use predicates, but probably not yet...

and add a one-line javadoc.

File: software/base/src/main/java/brooklyn/entity/brooklynnode/BrooklynNodeImpl.java [hunk 2]
```diff
@@ -81,9 +84,25 @@
         RendererHints.register(WEB_CONSOLE_URI, RendererHints.namedActionWithUrl());
     }
 
-    private class UnmanageThread extends Thread {
+    private static class UnmanageTask implements Runnable {
+        private Task<?> latchTask;
+        private Entity unmanageEntity;
+
+        public UnmanageTask(Task<?> latchTask, Entity unmanageEntity) {
```
Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/brooklynnode/BrooklynNodeImpl.java]: mark `latchTask` as `@Nullable`

**Comment by ahgittin** [comment]
good fixes; only minor comments then let's merge

**Comment by neykov** [comment]
@ahgittin Addressed comments.

**Pull request merged**